include/path_tracking_pid/details/second_order_lowpass.hpp

```diff
@@ -4,26 +4,41 @@
 
 namespace path_tracking_pid::details
 {
-// Error tracker for the last 3 error and filtered error values.
+/**
+ * @brief Discrete time second order lowpass filter
+ */
 class SecondOrderLowpass
 {
 public:
-  // Pushes the given value to the errors FIFO buffer. A corresponding filtered error value is calculated and pushed
-  // to the filtered errors FIFO buffer.
-  void push(double value);
+  /**
+   * @brief Construct a SecondOrderLowpass instance with NaNs
+   */
+  SecondOrderLowpass();
 
-  // Resets both errors and filtered errors FIFO buffers.
-  void reset();
+  /**
+   * @brief Construct a SecondOrderLowpass instance
+   * @param cutoff frequency in Hz, 0 disables the filter
+   * @param damping frequency in Hz
+   */
+  SecondOrderLowpass(double cutoff, double damping);
+
+  void configure(double cutoff, double damping);
```

Comment:
I see `configure()` is only used together with `reset()`. I can imagine that a `configure()` doesn't make sense without a `reset()`. If so, consider making the configure also do a reset.

Reply:
These functions have a different function. I'll document this better:
- configure sets the filter parameters. Can be done while driving.
- reset clears the internal buffer to 0. This should be done when starting a new path for example.

I've opened a ticket related to this: https://github.com/nobleo/path_tracking_pid/issues/121